src/main/java/io/nats/client/impl/NatsJetStreamPullSubscription.java

```diff
@@ -138,9 +138,10 @@ private List<Message> _fetch(int batchSize, long maxWaitMillis) {
             // timeout > 0 process as many messages we can in that time period
             // If we get a message that either manager handles, we try again, but
             // with a shorter timeout based on what we already used up
-            long timeLeft = maxWaitMillis;
-            while (batchLeft > 0 && timeLeft > 0) {
-                Message msg = nextMessageInternal( Duration.ofMillis(timeLeft) );
+            long maxWaitNanos = maxWaitMillis * 1_000_000;
+            long timeLeftNanos = maxWaitNanos;
```

Comment:
Shouldn't timeLeftNanos be initialized to maxWaitNanos - (System.nanoTime() - start) as well?

Reply:
I think the reasoning here is that we want to at least allow processing a single message to send back, which could no longer be the case if `timeLeftNanos` is set to `maxWaitNanos - (System.nanoTime() - start)` since we could set `maxWaitNanos` really small.